src/Entity/BannerSelection/Campaign.php

```diff
@@ -26,18 +24,18 @@ public function __construct(
 		string $identifier,
 		\DateTime $start,
 		\DateTime $end,
-		float $displayRatio,
-		RandomIntegerInterface $rng,
-		Bucket $mainBucket,
+		int $displayPercentage,
+		RandomIntegerGenerator $rng,
+		Bucket $firstBucket,
 		Bucket ...$additionalBuckets ) {
 
 		$this->identifier = $identifier;
 		$this->start = $start;
 		$this->end = $end;
-		$this->displayRatio = $displayRatio;
+		$this->displayPercentage = $displayPercentage;
 		$this->rng = $rng;
 		$this->buckets = $additionalBuckets;
-		array_unshift( $this->buckets, $mainBucket );
+		array_unshift( $this->buckets, $firstBucket );
```

Comment:
slightly inconsistent with `Bucket` & could potentially change the parameter by reference. I prefer the `array_merge` solution.